pkg/autoscaler/metrics/http_scrape_client.go

```diff
@@ -45,19 +48,42 @@ func (c *httpScrapeClient) Scrape(url string) (Stat, error) {
 	if err != nil {
 		return emptyStat, err
 	}
+
+	// Ask for protobuf by default. Note that during migration this will not trigger the proto format supported
+	// by Prometheus reporter as the latter uses `application/vnd.google.protobuf`.
+	req.Header.Add("Accept", "application/protobuf")
```

Comment:
Does this keep the auto-gzip handling for the prometheus handler intact or do we need to "manually" do that now?

Reply:
@markusthoemmes afaik the transport will ask for `gzip` by [itself.](https://golang.org/pkg/net/http/#Transport) when there is no accept-encoding header present. The Prometheus handler properly handles this in this [code part.](https://github.com/prometheus/client_golang/blob/b0cdec211b8bf9d5b1edabff4d32088a11f18310/prometheus/promhttp/http.go#L157) which checks if there is a [gzip encoding](https://github.com/prometheus/client_golang/blob/b0cdec211b8bf9d5b1edabff4d32088a11f18310/prometheus/promhttp/http.go#L356) in the hearder section. In the proto case gzip will not be triggered so that means we need to explicitly add gzip on in the request header before it is received by that handler.